opnode/rollup/driver/step.go

```diff
@@ -89,11 +89,12 @@ func (d *outputImpl) createNewBlock(ctx context.Context, l2Head eth.L2BlockRef,
 
 	// Next we append user deposits. If we're not the first block in an epoch, then receipts will
 	// be empty and no deposits will be derived.
-	deposits, err := derive.DeriveDeposits(receipts, d.Config.DepositContractAddress)
+	deposits, errs := derive.DeriveDeposits(receipts, d.Config.DepositContractAddress)
 	d.log.Info("Derived deposits", "deposits", deposits, "l2Parent", l2Head, "l1Origin", l1Origin)
-	if err != nil {
-		return l2Head, nil, fmt.Errorf("failed to derive deposits: %v", err)
+	for _, err := range errs {
+		d.log.Error("Failed to derive a deposit", "l1OriginHash", l1Origin.Hash, "err", err)
 	}
+	// TODO: Should we halt if len(errs) > 0? Opens up a denial of service attack, but prevents lockup of funds.
```

Comment:
Halting is better than losing funds in case of unexpected processing errors IMO. It should never happen though, so either way sounds fine, as long as we can catch this in testing. Curious what others think, but doesn't have to block this PR.